File: sources/julia/04_array_sum/main.jl
# task 04 array_sum — expected output: 499999500000
# build: julia main.jl    run: julia main.jl

function main()
    n = 1000000
    array = Vector{Int64}(undef, n)
    for i in 1:n
        array[i] = i - 1
    end

    total = Int64(0)
    for i in 1:n
        total += array[i]
    end
    println(total)
end

main()

File: sources/fsharp/04_array_sum/main.fs
// task 04 array_sum — expected output: 499999500000
// build: dotnet build -c Release    run: dotnet run

let n = 1000000
let arr : int64[] = Array.zeroCreate n

let mutable i = 0
while i < n do
    arr.[i] <- int64 i
    i <- i + 1

let mutable total = 0L
i <- 0
while i < n do
    total <- total + arr.[i]
    i <- i + 1

printfn "%d" total
